docs/v2-migration-guide.md

```diff
@@ -0,0 +1,105 @@
+# Version 2 Features
+
+Version 2 replaces the `Nullable` interface and various implementations
+(including user-defined implementations!) with just two generic types:
+`Update[T]` and `SliceUpdate[T]`.
+
+`Update` uses a value internally, rather than a pointer. This may reduce the
+need for allocations, and it also means that updates can now be meaningfully
+compared using `==`.
+
+Unlike `StringSlice`, `SliceUpdate` can distinguish between removed and set to
+`nil`. This is more in line with non-slice updates, which could always
+distinguish between removed and set to the zero value.
+
+In `nullable` version 1, only `nullable.String` implements `fmt.Stringer`. With
+go 1.18 generics, it's not possible to implement an interface only for
+particular type parameters of a generic type, leaving us with the choice to
+implement `fmt.Stringer` for all types or for none. We chose the former. The
+`Update` implementation uses a type switch to check if `T` is itself a `string`
+or `fmt.Stringer`, falling back to `fmt.Sprintf` with the `%v` verb for all
+other types. (The `SliceUpdate[T]` implementation always uses `fmt.Sprintf` with
+`%v` since a `[]T` is not `string` or `fmt.Stringer`.)
+
+# Migration Guide
+
+The package has been renamed from `nullable` to `nup` (for "nullable update").
+It's 62.5% shorter and 87% more whimsical! You should replace v1 imports
+(`github.com/nicheinc/nullable`) with `github.com/nicheinc/nullable/v2/nup`.
+
+The old `Value` method on each `Nullable` type has been replaced with three
+`Update` methods: `Value`, `ValueOperation`, and `ValueOrNil`:
```

Comment:
My rationale behind these three methods is to make value accesses more explicit and less prone to panics. `Value` now returns two values, and since unused local variables are an error, the user has to explicitly ignore the value with `_` to "assert" that the value exists, rather than just dereferencing a pointer. `ValueOperation` is similar. I originally wasn't going to include `ValueOrNil` at all, but it was too useful in our existing patterns (especially for passing updates as postgres parameters).